### DISCUSSION TITLE
First Class Types?

### DISCUSSION THREAD

**Discussion by ken-okabe** [uwu-tech/Kind#315] [Q&A] [answered]
https://idris2.readthedocs.io/en/latest/tutorial/typesfuns.html

>In Idris, types are first class, meaning that they can be computed and manipulated (and passed to functions) just like any other language construct. 

Is this possible in Kind?

**Comment by algebraic-dev** [comment, marked as answer]
Yes, e.g.
```kind
Choose (n: Bool) : Type
Choose Bool.true = Nat
Choose Bool.false = String
```
